### PR TITLE
Limit trading

### PR DESCRIPTION
**Pull request by Ali1**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**
Possibility to configure the maximum amount of fiat currency available to gekko to trade with

* **Other information**:
#937

Just an idea for an implantation. What do you think?

### CODE REVIEW

**Comment by greenbigfrog** [comment]
Looks fine (didn't test), but your commits are kinda messed up. Mike won't mind though most likely.

**Comment by inafets** [comment]
It would be nice to have the limit as a percentage of your funds

**Comment by Ali1** [comment]
@inafets could do. In your mind, is this a proportion of the initial currency balance when the trading bot is started?

**Comment by inafets** [comment]
@Ali1 What I think is to redefine the value of balance by the given proportion every time you are going to buy:`balance = this.getBalance(this.currency) * balancePercentage`

**Comment by Ali1** [comment]
But then if you buy 50% of your currency, and then advice comes in to buy again, what should happen?

File: plugins/trader/portfolioManager.js [hunk 1]
```diff
@@ -208,8 +225,9 @@ Manager.prototype.buy = function(amount, price) {
     'Attempting to BUY',
     amount,
     this.asset,
-    'at',
-    this.exchange.name,
```
Comment (Ali1) [inline plugins/trader/portfolioManager.js]: no longer necessary as logPortfolio is now being executed when trade advice is given which includes the name of the exchange.

File: sample-config.js [hunk 2]
```diff
@@ -216,7 +216,8 @@ config.trader = {
   key: '',
   secret: '',
   username: '', // your username, only required for specific exchanges.
-  passphrase: '' // GDAX, requires a passphrase.
+  passphrase: '', // GDAX, requires a passphrase.
+  currencyAmountAvailableToTrade = false; // set to a number to limit the amount of fiat currency available to Gekko to trade. Set to false to disable.
```
Comment (Ali1) [inline sample-config.js]: does this also need to go into https://github.com/askmike/gekko/blob/stable/config/plugins/trader.toml to set the defaults? not sure how these toml files work.

File: config/plugins/trader.toml [hunk 3]
```diff
@@ -6,4 +6,5 @@ enabled = false
 key = ""
 secret = ""
 # this is only required at specific exchanges
-username = ""
\ No newline at end of file
+username = ""
+currencyAmountAvailableToTrade = false
```
Comment (Ali1) [inline config/plugins/trader.toml:10]: is this needed? not sure how these toml files work

**Comment by Ali1** [comment]
This still doesn't work perfectly because the details of partially filled orders are not being tracked in Gekko. Any ideas?

**Comment by Ali1** [comment]
found a solution

**Comment by developdaly** [comment]
@Ali1, have you been running this PR successfully for a month now?

**Comment by Ali1** [comment]
It's been good except for one logging issue

image

I've applied a fix but waiting for relevant trades in the real world to show the result.

**Comment by Ali1** [comment]
@developdaly @askmike has been working great now

**Comment by Ali1** [comment]
Merged commits and resolved conflict in #1834

**Pull request closed without merging**